### DISCUSSION TITLE
How to catch exception thrown in NotifierProvider in an observer.

### DISCUSSION THREAD

**Discussion by TreyThomas93** [rrousselGit/riverpod#3462] [Q&A]
Hey everyone, I was hoping someone could help me out. I am using a provider observer to catch and log all of the failed providers, but when it comes to Notifier, if I throw an exception, then the observer doesn't catch it. Why is this and how can I catch the Notifier method exception in the observer? 

If this is not an ideal approach, then is better practice to just let the app crash and fail and log from there?

 ##### My current understanding
> FutureProvider fails, it automatically converts to AsyncError and observer picks it up.
AsyncNotifier fails, for which I have the method wrapped in try/catch, then I set state to AsyncError and the observer picks it up.
Provider fails and observer picks it up.
Notifier fails and DOESNT pick it up.

Is it possible to implement a riverpod class and create my own subclass to handle "SyncError" the same way AsyncError is handled?

Thanks

**Comment by rrousselGit** [comment]
What do you mean by the observer not picking up exceptions in Notifiers? It should work.
